doc/nrf/doc_styleguide.rst

```diff
@@ -305,27 +305,39 @@ Here is an example of a fully defined function:
 Enums
 =====
 
-The documentation block must precede the documented element.
-This is in accordance with the :ref:`Zephyr coding style <zephyr:coding_style>`.
+Place the documentation block above the enum.
+The documentation for the elements inside the enum can be above them or inline.
 
+An example with documentation preceding the documented element:
 
 .. code-block:: c
 
-        /** HID Service Protocol Mode events. */
-        enum hids_pm_evt {
+   /** HID Service Protocol Mode events. */
+   enum hids_pm_evt {
+
+      /** Boot mode entered. */
+      HIDS_PM_EVT_BOOT_MODE_ENTERED,
+
+      /** Report mode entered. */
+      HIDS_PM_EVT_REPORT_MODE_ENTERED,
+    };
 
-           /** Boot mode entered. */
-           HIDS_PM_EVT_BOOT_MODE_ENTERED,
+An example with the documentation inline (note the less than sign `<` after the asterisks `**`):
```

Comment:
We could link to https://www.doxygen.nl/manual/docblocks.html#memberdoc here , as this is not the only correct syntax for inline documentation after a member.

Reply:
Not sure whether it is feasible to give all the alternatives in the world of doxygen. If we link to this, we may end up having so many different variations in the header files.